src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -114,11 +114,38 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 
 		private IGossipSeed[] GetGossipCandidatesFromConfig() {
 			//_log.Debug("ClusterDnsEndPointDiscoverer: GetGossipCandidatesFromDns");
+			
+			IGossipSeed[] endpoints;
 
-			// Safe in this case.
-			IGossipSeed[] endpoints = _gossipSeeds;
+			if ((_gossipSeeds?.Length ?? 0) != 0) {
+				if (_compatibilityMode.IsAutoCompatibilityModeEnabled()) {
+					var tmp = new List<IGossipSeed>();
+					for (var i = 0; i != _gossipSeeds.Length; i++) {
```

Comment:
```suggestion
					for (var i = 0; i < _gossipSeeds.Length; i++) {
```
Just for safety reasons in case the code is modified later by someone else

Reply:
Done